summingbird-core/src/main/scala/com/twitter/summingbird/memory/Memory.scala

```diff
@@ -39,82 +41,85 @@ class Memory(implicit jobID: JobId = JobId("default.memory.jobId")) extends Plat
 
-  private def toStream[T](outerProducer: Prod[T], jamfs: JamfMap): (Stream[T], JamfMap) =
-    jamfs.get(outerProducer) match {
-      case Some(s) => (s, jamfs)
-      case None =>
-        val (s, m) = outerProducer match {
-          case NamedProducer(producer, _) => toStream(producer, jamfs)
-          case IdentityKeyedProducer(producer) => toStream(producer, jamfs)
-          case Source(source) => (source.toStream, jamfs)
-          case OptionMappedProducer(producer, fn) =>
-            val (s, m) = toStream(producer, jamfs)
-            (s.flatMap(fn(_)), m)
-
-          case FlatMappedProducer(producer, fn) =>
-            val (s, m) = toStream(producer, jamfs)
-            (s.flatMap(fn(_)), m)
-
-          case MergedProducer(l, r) =>
-            val (leftS, leftM) = toStream(l, jamfs)
-            val (rightS, rightM) = toStream(r, leftM)
-            (leftS ++ rightS, rightM)
-
-          case KeyFlatMappedProducer(producer, fn) =>
-            val (s, m) = toStream(producer, jamfs)
-            (s.flatMap {
-              case (k, v) =>
-                fn(k).map((_, v))
-            }, m)
-
-          case ValueFlatMappedProducer(producer, fn) =>
-            val (s, m) = toStream(producer, jamfs)
-            (s.flatMap {
-              case (k, v) =>
-                fn(v).map((k, _))
-            }, m)
-
-          case AlsoProducer(l, r) =>
-            //Plan the first one, but ignore it
-            val (left, leftM) = toStream(l, jamfs)
-            val (right, rightM) = toStream(r, leftM)
-            // We need to force all of left to make sure any
-            // side effects in write happen
-            lazy val lforcedEmpty = left.filter(_ => false)
-            (right.append(lforcedEmpty), rightM)
-
-          case WrittenProducer(producer, fn) =>
-            val (s, m) = toStream(producer, jamfs)
-            (s.map { i => fn(i); i }, m)
-
-          case LeftJoinedProducer(producer, service) =>
-            val (s, m) = toStream(producer, jamfs)
-            val joined = s.map {
-              case (k, v) => (k, (v, service.get(k)))
-            }
-            (joined, m)
-
-          case Summer(producer, store, semigroup) =>
-            val (s, m) = toStream(producer, jamfs)
-            val summed = s.map {
-              case (k, deltaV) =>
-                val oldV = store.get(k)
-                val newV = oldV.map { semigroup.plus(_, deltaV) }
-                  .getOrElse(deltaV)
-                store.update(k, newV)
-                (k, (oldV, deltaV))
-            }
-            (summed, m)
-        }
-        // scala can't infer that s is the right type through case statements above
-        val st = s.asInstanceOf[Stream[T]]
-        (st, m + (outerProducer -> st))
+  def toStream[T, K, V](outerProducer: Prod[T], jamfs: JamfMap): (Stream[T], JamfMap) = {
+    val stream = jamfs.get(outerProducer)
```

Comment:
could we do: `Option(jamfs.get(outerProducer)) match {` so that the diff can be smaller? This is harder to read.